shopinvader_membership/services/membership.py

```diff
@@ -39,6 +41,41 @@ def search(self, **params):
         """
         return self._paginate_search(**params)
 
+    def subscribe(self, _id):
+        """
+        Subscribe to a membership product with logged user
+        :param _id: id of product.product
+        :return: dict with invoice_id
+        """
+        if not self._is_logged():
+            raise UserError(_("A user should be logged"))
+        membership_product = self.env["product.product"].search(
+            [("id", "=", _id), ("membership", "=", True)]
```

Comment:
I think you should add `limit=1` because there I don't see any constrain who says you can have only 1 membership product. (or maybe you should raise an exception if more than 1 product found?)

Reply:
@acsonefho how should he find more than 1 product with the id in the domain ?